pkg/bootstrap/config.go

```diff
@@ -154,7 +155,7 @@ func (cfg Config) toTemplateParams() (map[string]any, error) {
 				opts = append(opts,
 					option.Localhost(option.LocalhostIPv4),
 					option.Wildcard(option.WildcardIPv4),
-					option.DNSLookupFamily(option.DNSLookupFamilyIPS))
+					option.DNSLookupFamily(option.DNSLookupFamilyIPv4Pred))
```

Comment:
Hi @howardjohn, it's different because only `DnsLookupFamily` should be set in `cluster_builder.go`, but both `Localhost` and `Wildcard` should be set here, the specified IP format should be confirmed.

Reply:
Hi @howardjohn, I think we may need to only focus on flag branch, if so , the cluster_builder use `ALL` and we use `ALL` or `V4_PREFERRED` in bootstrap. In fact I have tried below in bootstrap: 

```
if features.EnableDualStack {
	opts = append(opts,
		option.Localhost(option.LocalhostIPv6),
		option.Wildcard(option.WildcardIPv6),
		option.DNSLookupFamily(option.DNSLookupFamilyIPS))
	opts = append(opts,
		option.Localhost(option.LocalhostIPv4),
		option.Wildcard(option.WildcardIPv4),
		option.DNSLookupFamily(option.DNSLookupFamilyIPS))
}
```
And it does not work. Frankly, I do not know why, but I doubt that there may be bugs for `Cluster_ALL` in envoy. However, I am trying as below to remove the `CheckIPFamilyTypeForFirstIPs` checking.

Reply:
Hi @howardjohn, I think we may need to only focus on flag branch, if so , the cluster_builder use `ALL` and we use `ALL` or `V4_PREFERRED` in bootstrap. In fact I have tried below in bootstrap: 

```
if features.EnableDualStack {
	opts = append(opts,
		option.Localhost(option.LocalhostIPv6),
		option.Wildcard(option.WildcardIPv6),
		option.DNSLookupFamily(option.DNSLookupFamilyIPS))
	opts = append(opts,
		option.Localhost(option.LocalhostIPv4),
		option.Wildcard(option.WildcardIPv4),
		option.DNSLookupFamily(option.DNSLookupFamilyIPS))
}
```
And it does not work. Frankly, I do not know why, but I doubt that there may be bugs for `Cluster_ALL` in envoy. However, I am trying as below to remove the `CheckIPFamilyTypeForFirstIPs` checking.

```
if features.EnableDualStack {
	opts = append(opts,
		option.Localhost(option.LocalhostIPv6),
		option.Wildcard(option.WildcardIPv6),
		option.DNSLookupFamily(option.DNSLookupFamilyIPS))
	opts = append(opts,
		option.Localhost(option.LocalhostIPv4),
		option.Wildcard(option.WildcardIPv4),
		option.DNSLookupFamily(option.DNSLookupFamilyIPv4Pred))
}
```

Reply:
Hi @howardjohn, I think we may need to only focus on flag branch, if so , the cluster_builder use `ALL` and we use `ALL` or `V4_PREFERRED` in bootstrap. 
I doubt that there may be bugs for `Cluster_ALL` in envoy. However, I am trying as below to remove the `CheckIPFamilyTypeForFirstIPs` checking.

```
if features.EnableDualStack {
	opts = append(opts,
		option.Localhost(option.LocalhostIPv6),
		option.Wildcard(option.WildcardIPv6),
		option.DNSLookupFamily(option.DNSLookupFamilyIPS))
}
```

Reply:
Hi @howardjohn, I think the `CheckIPFamilyTypeForFirstIPs ` should be in bootstrap, however, it's okay if I can change it as below:
```
if features.EnableDualStack {
                        // If dual-stack, it may be [IPv4, IPv6] or [IPv6, IPv4]
                        // So let the first ip family policy to decide its DNSLookupFamilyIP policy
                        ipFamily := network.CheckIPFamilyTypeForFirstIPs(cfg.Metadata.InstanceIPs)
                        if ipFamily == network.IPv6 {
                                opts = append(opts,
                                        option.Localhost(option.LocalhostIPv6),
                                        option.Wildcard(option.WildcardIPv6),
                                        option.DNSLookupFamily(option.DNSLookupFamilyIPS))
                        } else {
                                opts = append(opts,
                                        option.Localhost(option.LocalhostIPv4),
                                        option.Wildcard(option.WildcardIPv4),
                                        option.DNSLookupFamily(option.DNSLookupFamilyIPS))
                        }
                } else {
...
```

At least, I think it uniforms for using `Cluster_ALL` in both `cluster builder` and `bootstrap`, does it make sense?

Reply:
I think it definitely 100% needs to be the same as cluster_builder.go. Beyond that I don't have enough context.

Reply:
Hi @howardjohn, okay, I got what your mean, both `cluster builder` and `bootstrap` are using `Cluster_ALL` now, and I also removed the `V4_PREFERRED` because it's not used.